server/provider/constants.go

```diff
@@ -5,6 +5,7 @@ var (
 	genesisBlockNonce         = 0
 	// Block with nonce = 0 is not actually retrievable from the observer
 	oldestPossibleNonceWithHistoricalState = uint64(1)
+	blocksCacheCapacity                    = 256
```

Comment:
For a scenario with very large blocks (including transactions and their logs): ~ 4 MB each * 256 = 1 GB (a low memory footprint for regular observers).